### PR TITLE
feat(web): feature/BOAT-851-start-case-date-flow

### PR DESCRIPTION
**Pull request by kandcdev**
## Describe your changes
BOAT-851

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: appeals/web/src/server/appeals/appeal-details/start-case/__tests__/__snapshots__/start-case.test.js.snap [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`start-case GET /start-case/date should render "Case started" confirmation page should 1`] = `
```
Comment (james-hill-kc) [inline appeals/web/src/server/appeals/appeal-details/start-case/__tests__/__snapshots__/start-case.test.js.snap:3]: Is the route/path correct here? I'd expect the path to be something like `/confirmation` for a confirmation screen. Also there seems to be a typo (extra `should` at the end)

File: appeals/web/src/server/appeals/appeal-details/start-case/__tests__/start-case.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import nock from 'nock';
+import supertest from 'supertest';
+import { createTestEnvironment } from '#testing/index.js';
+import { appealData } from '#testing/appeals/appeals.js';
+
+const { app, teardown } = createTestEnvironment();
+const request = supertest(app);
+const baseUrl = '/appeals-service/appeal-details';
+
+describe('start-case', () => {
```
Comment (james-hill-kc) [inline appeals/web/src/server/appeals/appeal-details/start-case/__tests__/start-case.test.js:11]: Ideally there should be unit tests for the `POST` route as well as the `GET`

File: appeals/web/src/server/appeals/appeal-details/start-case/__tests__/start-case.test.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+			expect(response.text).toContain('Confirming will activate the timetable on');
+		});
+
+		it('should redirect to the confirmation page', async () => {
+			nock('http://test/').patch(`/appeals/1`).reply(200, {});
+			const response = await request.post(`${baseUrl}/1/start-case/date`).send({});
+			expect(response.statusCode).toBe(302);
+		});
+
+		it('should render "Case started" confirmation page should', async () => {
```
Comment (james-hill-kc) [inline appeals/web/src/server/appeals/appeal-details/start-case/__tests__/start-case.test.js:29]: This should be moved into a separate `describe` - this one is just for `GET /start-case/date`. Same for the `POST` test above.

**Review by james-hill-kc — COMMENTED** [review]
Looking good overall, I've left a few comments

File: appeals/web/src/server/appeals/appeal-details/start-case/__tests__/start-case.test.js (hunk 2 above)
Comment (kandcdev) [reply appeals/web/src/server/appeals/appeal-details/start-case/__tests__/start-case.test.js:11]: Yes, but in this case user is not posting any data. The data is being posted is auto-generated "today" date

File: appeals/web/src/server/appeals/appeal-details/start-case/__tests__/__snapshots__/start-case.test.js.snap (hunk 1 above)
Comment (kandcdev) [reply appeals/web/src/server/appeals/appeal-details/start-case/__tests__/__snapshots__/start-case.test.js.snap]: done

File: appeals/web/src/server/appeals/appeal-details/start-case/__tests__/start-case.test.js (hunk 3 above)
Comment (kandcdev) [reply appeals/web/src/server/appeals/appeal-details/start-case/__tests__/start-case.test.js]: done

File: appeals/web/src/server/appeals/appeal-details/start-case/__tests__/start-case.test.js (hunk 2 above)
Comment (james-hill-kc) [reply appeals/web/src/server/appeals/appeal-details/start-case/__tests__/start-case.test.js:11]: On second look, there is actually a `POST` test - `should redirect to the confirmation page` - ideally that should be in its own `describe` but it's very minor

**Review by james-hill-kc — APPROVED** [review]

**Pull request merged**